modules/api/php/models/projectimagesrow.class.inc

```diff
@@ -74,12 +74,13 @@ class ProjectImagesRow implements \LORIS\Data\DataInstance
             'QC_status'  => $this->_qcstatus,
             'Selected'   => $this->_selected,
         ];
-        $filename = basename($this->_filename);
+        $filename = basename($this->_filepath);
         $candid   = $obj['Candidate'];
         $visit    = $obj['Visit'];
 
         $obj['Link']       = "/candidates/$candid/$visit/images/$filename";
         $obj['InsertTime'] = date('c', (int) $obj['InsertTime']);
+        $obj['File']       = $filename;
```

Comment:
This was specifically removed in #6516 and should not be added without a new version of the spec. (In fact, it shouldn't be added at all.. it's just leaking internal details and the file is available to API users through the link field)

Reply:
In this case, adding the filename doesn't leak any additional details, the filename in https://demo.loris.ca/api/v0.0.3/projects/pumpernickel/images  is already available in the `Link` , so it is still leaking the file's name. I thought it might be redundant, but I tried to make it consistent with https://demo.loris.ca/api/v0.0.3/projects/pumpernickel/recordings , where the field `File` was actually the path.

Reply:
I don't mean it's leaking sensitive information, I mean it's leaking implementation details. The path on the filesystem is irrelevant for users of the REST API and unusable without access to the backend (in which case there's no reason to be using the REST API in the first place.) There is already a mechanism in the API to get the data for the file through the API -- the `Link` attribute, which doesn't require a leaky abstraction.

Reply:
I know that the path is irrelevant to the user, that is actually what I am trying to change in  `modules/api/php/models/projectrecordingsrow.class.inc` . In this case, `File` doesn't mean the file's path, but the file's name. I can change it for `Filename` to be more precise. In the code, `File` keeps is referred to as the file's name (e.g. `$this->_filename   = $row['File'] ?? null;`), but it's actually the file path, as you can see in https://demo.loris.ca/api/v0.0.3/projects/pumpernickel/recordings (which is the same thing on my VM on the `aces/main` branch).

Reply:
I know very little about the EEG part of the API, Cecile would need to provide feedback on that, but if you're changing it to something other than what the spec currently says you'd need to create a new version/update the spec for that new version.

From what I can see on the demo link that you provided, it looks like the "sub-OTT167/ses-V1/eeg/" part may be relevant for the BIDS standard and isn't in the Link. It might make sense to remove the `File` and add something like `BIDSName` with a value like `sub-OTT167/ses-V1/eeg/sub-OTT167_ses-V1_task-faceO_eeg.edf`, but @cmadjar would need to need to weigh in on that.